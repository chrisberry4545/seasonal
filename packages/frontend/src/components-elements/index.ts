export * from './BackgroundImageWithLoadingSpinner/BackgroundImageWithLoadingSpinner';

export * from './GroupedSelectBox/GroupedSelectBox';
