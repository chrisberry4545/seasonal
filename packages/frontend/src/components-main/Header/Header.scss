@import './../../../node_modules/@chrisb-dev/seasonal-shared-frontend-components/dist/styles/settings';
@import './../../../node_modules/@chrisb-dev/seasonal-shared-frontend-components/dist/styles/tools';

.c-header {
  display: flex;
  align-items: center;
  border-bottom: 1px solid $grey-med;
  margin: 0 0.8rem;
  position: fixed;
  height: 60px;
  top: 0;
  left: 0;
  right: 0;
  background-color: $background-color;
  z-index: 1;
}

  .c-header__btn {
    width: 30px;
    height: 100%;
    display: flex;
    align-items: center;
    position: absolute;
    top: 50%;
    left: 0;
    transform: translateY(-50%);
    padding: 0;
    @include mq($from: tablet) {
      display: none;
    }
  }

  .c-header__text {
    margin: 0 3rem;
    width: 100%;
    text-align: center;
  }
