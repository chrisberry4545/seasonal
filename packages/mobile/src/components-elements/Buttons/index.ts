export * from './BareButton';

export * from './BorderedButton';
