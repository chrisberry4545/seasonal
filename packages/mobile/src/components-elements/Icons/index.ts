export * from './ArrowIcon';

export * from './BurgerIcon';

export * from './CrossIcon';

export * from './GridIcon';

export * from './ListIcon';

export * from './MagnifyingGlassIcon';
