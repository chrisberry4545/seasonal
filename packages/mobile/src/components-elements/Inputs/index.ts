export * from './Input';

export * from './MultilineInput';

export * from './RadioButton';

export * from './RadioButtonGroup';
