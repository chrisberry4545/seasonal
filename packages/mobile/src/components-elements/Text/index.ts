export * from './TextHeadingLarge';

export * from './TextHeadingMedium';

export * from './TextHeadingSmall';

export * from './TextLarge';

export * from './TextMedium';

export * from './TextSmall';
