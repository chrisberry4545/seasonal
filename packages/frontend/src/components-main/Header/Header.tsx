import React, { FC } from 'react';

import './Header.scss';

import {
  BareButton,
  BurgerIcon,
  TextHeadingLarge
} from '@chrisb-dev/seasonal-shared-frontend-components';
import {
  SearchBarConnecter
} from '../';
import { IHeaderDispatchProps } from './Header.interface';

export const Header: FC<IHeaderDispatchProps> = ({
  onMenuOpen
}) => (
  <header className='c-header'>
    <BareButton className='c-header__btn' onClick={onMenuOpen}>
      <BurgerIcon />
    </BareButton>
    <h1 className='c-header__text'>
      <TextHeadingLarge>
        Eat Seasonal
      </TextHeadingLarge>
    </h1>
    <SearchBarConnecter />
  </header>
);
