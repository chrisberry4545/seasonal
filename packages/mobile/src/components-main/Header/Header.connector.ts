import { connect } from 'react-redux';
import {
  Header
} from './Header';
import {
  openMenu, closeMenu
} from '@chrisb-dev/seasonal-shared-models';
import { IHeaderDispatchProps } from './Header.interface';
import { Dispatch } from 'redux';

const mapStateToProps = () => ({});

const mapDispatchToProps = (
  dispatch: Dispatch
): IHeaderDispatchProps => ({
  onMenuClose: () => dispatch(closeMenu()),
  onMenuOpen: () => dispatch(openMenu())
});

export const HeaderConnecter = connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
