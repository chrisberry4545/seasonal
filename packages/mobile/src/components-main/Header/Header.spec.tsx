import React from 'react';
import { Header } from './Header';
import { shallow, ShallowWrapper } from 'enzyme';
import { NavigationEvents, NavigationEventPayload } from 'react-navigation';
import { BareButton } from '../../components-elements';

jest.mock('../../components-elements', () => ({
  BareButton: () => 'BareButton',
  BurgerIcon: () => 'BurgerIcon',
  TextHeadingLarge: () => 'TextHeadingLarge'
}));

describe('<Header />', () => {
  let wrapper: ShallowWrapper;
  let mockOnMenuOpen: jest.Mock;
  let mockOnMenuClose: jest.Mock;

  beforeEach(() => {
    mockOnMenuOpen = jest.fn();
    mockOnMenuClose = jest.fn();
    wrapper = shallow(
      <Header
        onMenuClose={mockOnMenuClose}
        onMenuOpen={mockOnMenuOpen} />
    );
  });

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());

  test('closes the menu when opened', () => {
    const onDidFocus = wrapper.find(NavigationEvents).first().props().onDidFocus;
    if (onDidFocus) {
      onDidFocus({} as NavigationEventPayload);
    }
    expect(mockOnMenuClose).toHaveBeenCalled();
  });

  test('can open the menu', () => {
    const onClick = wrapper.find(BareButton).first().props().onClick;
    if (onClick) {
      onClick();
    }
    expect(mockOnMenuOpen).toHaveBeenCalled();
  });
});
