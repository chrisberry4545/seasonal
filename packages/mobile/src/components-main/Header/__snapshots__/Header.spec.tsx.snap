// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomWidth": 0.5,
      "borderColor": "#CCCCCC",
      "flexDirection": "row",
      "paddingBottom": 10,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 16,
    }
  }
>
  <BareButton
    onClick={[MockFunction]}
    style={
      Object {
        "flex": 0,
        "paddingTop": 4,
      }
    }
  >
    <BurgerIcon
      size={32}
    />
  </BareButton>
  <TextHeadingLarge
    style={
      Object {
        "flex": 1,
        "textAlign": "center",
      }
    }
  >
    Eat Seasonal
  </TextHeadingLarge>
  <Connect(Component) />
</View>
`;
